src/controllers/moderationactions/ModerationAction.cpp

```diff
@@ -140,18 +122,18 @@ const std::optional<ImagePtr> &ModerationAction::getImage() const
 {
     assertInGuiThread();
 
-    if (this->imageToLoad_ != 0)
+    if (this->builtInImageToLoad_ == BuiltInImage::Ban)
     {
-        if (this->imageToLoad_ == 1)
-        {
-            this->image_ =
-                Image::fromResourcePixmap(getResources().buttons.ban);
-        }
-        else if (this->imageToLoad_ == 2)
-        {
-            this->image_ =
-                Image::fromResourcePixmap(getResources().buttons.trashCan);
-        }
+        this->image_ = Image::fromResourcePixmap(getResources().buttons.ban);
+    }
+    else if (this->builtInImageToLoad_ == BuiltInImage::TrashCan)
+    {
+        this->image_ =
+            Image::fromResourcePixmap(getResources().buttons.trashCan);
+    }
+    else if (!this->iconPath_.isEmpty())
+    {
+        this->image_ = Image::fromUrl({this->iconPath_.toString()});
     }
```

Comment:
It was like this before, but the image should probably be set once (even though there's a cache).

Reply:
I added an early-return